pyccel/errors/messages.py

```diff
@@ -24,6 +24,8 @@
 INCOMPATIBLE_REDEFINITION = 'Incompatible redefinition'
 INCOMPATIBLE_REDEFINITION_STACK_ARRAY = 'Cannot change shape of stack array, because it does not support memory reallocation. Avoid redefinition, or use standard heap array.'
 STACK_ARRAY_DEFINITION_IN_LOOP = 'Cannot create stack array in loop, because if does not support memory reallocation. Create array before loop, or use standard heap array.'
+STACK_ARRAY_NON_LITERAL_SHAPE = 'Cannot create stack array from a shape which is unknown at function entry'
```

Comment:
I suggest to change the name of this variable, as it is a bit confusing for the developer. Maybe `STACK_ARRAY_UNKNOWN_SHAPE`?

Reply:
Fixed [b577d1a](https://github.com/pyccel/pyccel/pull/993/commits/b577d1afaeb07202a2e4bdab122a84b514c50572)